frontendExternalSystem/src/main/java/com/virtuslab/gitmachete/frontend/externalSystem/project/MacheteProjectResolver.java

```diff
@@ -0,0 +1,55 @@
+package com.virtuslab.gitmachete.frontend.externalSystem.project;
+
+import java.io.File;
+
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.externalSystem.model.DataNode;
+import com.intellij.openapi.externalSystem.model.ExternalSystemException;
+import com.intellij.openapi.externalSystem.model.ProjectKeys;
+import com.intellij.openapi.externalSystem.model.project.ProjectData;
+import com.intellij.openapi.externalSystem.model.task.ExternalSystemTaskId;
+import com.intellij.openapi.externalSystem.model.task.ExternalSystemTaskNotificationListener;
+import com.intellij.openapi.externalSystem.service.project.ExternalSystemProjectResolver;
+import com.intellij.ui.GuiUtils;
+import io.vavr.control.Option;
+import lombok.CustomLog;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.externalSystem.MacheteProjectService;
+import com.virtuslab.gitmachete.frontend.externalSystem.settings.MacheteExecutionSettings;
+import com.virtuslab.gitmachete.frontend.ui.api.table.GraphTableProvider;
+
+@CustomLog
+public class MacheteProjectResolver implements ExternalSystemProjectResolver<MacheteExecutionSettings> {
+
+  @Override
+  public DataNode<ProjectData> resolveProjectInfo(ExternalSystemTaskId id,
+      String projectPath,
+      boolean isPreviewMode,
+      @Nullable MacheteExecutionSettings settings,
+      ExternalSystemTaskNotificationListener listener)
+      throws ExternalSystemException, IllegalArgumentException, IllegalStateException {
+    LOG.info("Performing");
+
+    var graphTableProvider = Option.of(settings)
+        .map(s -> s.getProject().getService(GraphTableProvider.class));
+
+    GuiUtils.invokeLaterIfNeeded(() -> {
+      if (graphTableProvider.isEmpty()) {
```

Comment:
Nit: maybe just no need to `invokeLater` at all if `graphTableProvider` is empty?